core/embed/bootloader/messages.c

```diff
@@ -707,6 +709,26 @@ int process_msg_FirmwareUpload(uint8_t iface_num, uint32_t msg_size,
     return UPLOAD_ERR_FIRMWARE_TOO_BIG;
   }
 
+  // check padding between headers and the code
+  for (size_t i = headers_end; i < headers_offset; i++) {
+    if (CHUNK_BUFFER_PTR[i] != 0) {
+      if (firmware_upload_chunk_retry > 0) {
```

Comment:
What's the purpose of the block beginning with `if (firmware_upload_chunk_retry > 0 ) {`? Is it really needed?

Reply:
the purpose is to retry uploading the chunk if the check fails, so that user doesn't have to manually retry if it fails for any reason. Its also done for hash check below. Not strictly necessary though.

Reply:
I get it. Even though USB packets have a CRC field, it's only 16-bit, so there's a small chance a packet could still arrive corrupted. Personally, I think that risk is pretty low, and I'd prefer to keep the code simpler since this FSM is already quite complex. But I’m not set on this, so it's totally up to you.